### PR TITLE
Auto configure all test tasks

### PR DESCRIPTION
**Pull request by alpar-t**
With this change, we apply the common test config automatically to all
newly created test tasks instead of opting in specifically.

For plugin authors using the plugin externally this means that the
configuration will be applied to their RandomizedTestingTasks as well.

The purpose of the task is to simplify setup and make it easier to
change projects that use the `test` task but actually run integration
tests to use a task called `integTest` for clarity, but also because
we may want to configure and run them differently.
E.x. using different levels of concurrency.

### CODE REVIEW

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy [hunk 1]
```diff
@@ -776,9 +778,8 @@ class BuildPlugin implements Plugin<Project> {
         }
     }
 
-    /** Returns a closure of common configuration shared by unit and integration tests. */
-    static Closure commonTestConfig(Project project) {
-        return {
+    static void applyCommonTestConfigPresentAndFuture(Project project) {
```
Comment (rjernst) [inline buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy]: We have other methods here which apply to all existing and future items of a type. I don't think we need such a verbose name for these cases. `configureTests` should be sufficient?

**Review by rjernst — APPROVED** [review]
LGTM. One minor naming ask.

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy (hunk 1 above)
Comment (atorok) [reply buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy]: In retrospect that does seem excessive. 
I think there is some value in coming up with something other than `apply` and `configure` to distinguish between what configures the current state vs attaching configuration to the domain object containers that affect future state.
For me this is one of the top confusions and sources of bugs, but we can do this in a follow up, I simplified the name for now.

**Comment by atorok** [comment]
@elasticmachine test this please

**Pull request merged**